src/task/ReceiveVideoStreamIndexTask.ts

```diff
@@ -74,13 +74,20 @@ export default class ReceiveVideoStreamIndexTask extends BaseTask
       videoUplinkBandwidthPolicy,
     } = this.context;
 
+    const oldAttendees = videoStreamIndex.allVideoSendingAttendeesExcludingSelf(selfAttendeeId);
     videoStreamIndex.integrateIndexFrame(indexFrame);
     videoDownlinkBandwidthPolicy.updateIndex(videoStreamIndex);
     videoUplinkBandwidthPolicy.updateIndex(videoStreamIndex);
 
     this.resubscribe(videoDownlinkBandwidthPolicy, videoUplinkBandwidthPolicy);
     this.updateVideoAvailability(indexFrame);
     this.handleIndexVideosPausedAtSource();
+    const newAttendees = videoStreamIndex.allVideoSendingAttendeesExcludingSelf(selfAttendeeId);
+    if (oldAttendees.length !== newAttendees.length) {
```

Comment:
Doesn't it just check length here and not call the observer unless the lengths are different?
We need to do a deeper inspection to see if the attendee's themselves are different.

Reply:
Yes, yes I am checking this one after Kyu's comment, and updating it. Thanks!